IPA-SRA makes debugging hard, so do not enable at -O2 until that is fixed.

Index: gcc/opts.c
===================================================================
*** gcc/opts.c	(revision 159568)
--- gcc/opts.c	(working copy)
*************** decode_options (unsigned int argc, const
*** 906,912 ****
    flag_tree_pre = opt2;
    flag_tree_switch_conversion = opt2;
    flag_ipa_cp = opt2;
-   flag_ipa_sra = opt2;
  
    /* Track fields in field-sensitive alias analysis.  */
    set_param_value ("max-fields-for-field-sensitive",
--- 906,911 ----
*************** decode_options (unsigned int argc, const
*** 926,931 ****
--- 925,931 ----
    flag_ipa_cp_clone = opt3;
    if (flag_ipa_cp_clone)
      flag_ipa_cp = 1;
+   flag_ipa_sra = opt3;
  
    /* Just -O1/-O0 optimizations.  */
    opt1_max = (optimize <= 1);
